fmriprep/workflows/bold/registration.py

```diff
@@ -362,14 +385,25 @@ def init_bbreg_wf(
             (mri_coreg, transforms, [('out_lta_file', 'in2')]),
```

Comment:
Here you're still using `mri_coreg`. One problem here is going to be that `t2w_preproc` may be off from `T1.mgz` by `fsnative2t1w_xfm`.

I think it might be cleaner to change the `--bold2t1w-init <register|header>` option to `--bold2anat-init <auto|t1w|t2w|header>`. If we get `auto`, then we look for a T2w image in the inputs and assume `T2.mgz` will be present in the FreeSurfer directory (we can always improve this check later). Then we add a `FreeSurferSource` here to find `T2.mgz` and pass it to `mri_coreg` if `bold2anat_init == 't2w'`.